src/commands/pets/PetTransferCommand.ts

```diff
@@ -0,0 +1,165 @@
+	}));
+	return interaction.reply({embeds: [confirmEmbed]});
+}
+
+function sendErrorInvalidPositionShelter(guildPetCount: number, interaction: CommandInteraction, language: string, petTransferModule: TranslationModule) {
+	if (guildPetCount === 1) {
+		return sendErrorMessage(interaction.user, interaction.channel, language, petTransferModule.get("wrongPetNumberSingle"), false, interaction);
+	}
+	return sendErrorMessage(interaction.user, interaction.channel, language, petTransferModule.format("wrongPetNumberBetween", {
+		max: guildPetCount
+	}), false, interaction);
+}
+
+async function switchPets(guild: Guild, shelterPosition: any, interaction: CommandInteraction, language: string, petTransferModule: TranslationModule, entity: Entity) {
+	const pPet = entity.Player.Pet;
+	const swPet = guild.GuildPets[shelterPosition - 1];
+	const swPetEntity = swPet.PetEntity;
+
+	if (pPet) {
+		if (pPet.isFeisty()) {
+			sendErrorMessage(interaction.user, interaction.channel, language, petTransferModule.get("isFeisty"), false, interaction);
+			return null;
+		}
+		swPet.petEntityId = pPet.id;
+		await swPet.save();
+	}
+	else {
+		await swPet.destroy();
+	}
+	entity.Player.petId = swPetEntity.id;
+	await entity.Player.save();
+	return swPetEntity;
+}
+
+function setDescriptionPetTransferEmbed(pPet: PetEntity, confirmEmbed: DraftBotEmbed, petTransferModule: TranslationModule, language: string, swPetEntity: PetEntity) {
+	if (pPet) {
+		confirmEmbed.setDescription(petTransferModule.format("confirmSwitch", {
+			pet1: pPet.getPetEmote() + " " + (pPet.nickname ? pPet.nickname : pPet.getPetTypeName(language)),
+			pet2: swPetEntity.getPetEmote() + " " + (swPetEntity.nickname ? swPetEntity.nickname : swPetEntity.getPetTypeName(language))
+		}));
+	}
+	else {
+		confirmEmbed.setDescription(petTransferModule.format("confirmFollows", {
+			pet: swPetEntity.getPetEmote() + " " + (swPetEntity.nickname ? swPetEntity.nickname : swPetEntity.getPetTypeName(language))
+		}));
+	}
+}
+
+async function updateMissionsOfEntity(entity: Entity, interaction: CommandInteraction, language: string, swPetEntity: PetEntity) {
+	await MissionsController.update(entity.discordUserId, interaction.channel, language, "havePet");
+	await MissionsController.update(entity.discordUserId, interaction.channel, language, "tamedPet", 1, {loveLevel: swPetEntity.getLoveLevelNumber()});
+	await MissionsController.update(entity.discordUserId, interaction.channel, language, "trainedPet", 1, {loveLevel: swPetEntity.getLoveLevelNumber()});
+}
+
+export const commandInfo: ICommand = {
+	slashCommandBuilder: new SlashCommandBuilder()
+		.setName("pettransfer")
+		.setDescription("Leave your pet in the guild shelter and optionnally take one back")
+		.addIntegerOption(option => option.setName("shelterposition")
+			.setDescription("The position of the pet in the shelter you want to switch with")
+			.setRequired(false)
+			.setMinValue(1)
```

Comment:
Si il y a 0 pet dans le shelter, l'option devrait pas accepter de valeur non ?

Reply:
le problème ici est que l'on peut mettre que des variables dites statiques, rien de dynamique (comme pour les tableaux en C).

Reply:
I see.. that's sad